src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React, { PureComponent } from 'react';
+import { withSize } from '../shared/WithSize';
+import DivWithTooltip from '../shared/DivWithTooltip';
+
+import type { Thread } from '../../types/profile';
+import type { Milliseconds, StartEndRange } from '../../types/units';
+
+import './EmptyThreadIndicator.css';
+
+type SyntheticCssDeclarations = {
+  [string]: string | number,
+};
+
+type Props = {|
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: number,
+  +thread: Thread,
+  +interval: Milliseconds,
+  +unfilteredSamplesRange: StartEndRange | null,
+|};
+
+class EmptyThreadIndicator extends PureComponent<Props> {
+  _canvas: HTMLCanvasElement | null;
+  _requestedAnimationFrame: boolean | null;
+
+  _getIndicatorPositions(): {|
+    startup: SyntheticCssDeclarations | null,
+    shutdown: SyntheticCssDeclarations | null,
+    emptyBufferStart: SyntheticCssDeclarations | null,
+  |} {
+    const {
+      rangeStart,
+      rangeEnd,
+      width,
+      unfilteredSamplesRange,
+      interval,
+      thread: { processShutdownTime, registerTime, unregisterTime },
+    } = this.props;
+    const rangeLength = rangeEnd - rangeStart;
+    const xPixelsPerMs = width / rangeLength;
+
+    const threadEndTime =
+      unregisterTime === null ? processShutdownTime : unregisterTime;
+
+    // Did this thread startup in this time range?
+    let startup = null;
+    if (registerTime > rangeStart) {
+      startup = {
+        left: 0,
+        width: (registerTime - rangeStart) * xPixelsPerMs,
+      };
+    }
+
+    // Did this thread shut down in this time range?
+    let shutdown = null;
+    if (threadEndTime !== null && threadEndTime < rangeEnd) {
+      shutdown = {
+        right: 0,
+        width: (rangeEnd - threadEndTime) * xPixelsPerMs,
+      };
+    }
+
+    let emptyBufferStart = null;
+    if (
+      // Threads could have no samples, and therefore no range..
+      unfilteredSamplesRange !== null &&
+      // Was the buffer empty at the beginning of the range?
+      unfilteredSamplesRange.start > rangeStart &&
+      // Provide an interval's length of leeway to not show the indicator.
+      unfilteredSamplesRange.start - rangeStart > interval
```

Comment:
(rewriting this comment because the same one in my previous review has been "outdated")

I believe that if this is true then the previous condition is true too. (at least if interval is > 0). So the first condition is useless. Or I miss something ;)